pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -248,9 +275,47 @@ func validateRuleWithOperations(ruleWithOperations *admissionregistration.RuleWi
 }
 
 func ValidateValidatingWebhookConfigurationUpdate(newC, oldC *admissionregistration.ValidatingWebhookConfiguration) field.ErrorList {
-	return ValidateValidatingWebhookConfiguration(newC)
+	allErrors := genericvalidation.ValidateObjectMeta(&newC.ObjectMeta, false, genericvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+
+	oldWebhooks := map[string]admissionregistration.Webhook{}
+	for _, hook := range oldC.Webhooks {
+		oldWebhooks[hook.Name] = hook
+	}
+
+	for i, hook := range newC.Webhooks {
+		fldPath := field.NewPath("webhooks").Index(i)
+		allErrors = append(allErrors, validateWebhook(&hook, fldPath)...)
+		oldHook, exists := oldWebhooks[hook.Name]
+		if exists {
+			// Only validate admissionReviewVersions if they were valid before
```

Comment:
skipping this if the old webhook didn't have recognized versions means we also skip required and duplicate checking, which isn't great

Reply:
it might be simpler to do this at the object level rather than the webhook level (being more fine-grained doesn't do a lot for us, and makes the code way more complicated):

* set hadRecognizedVersions to true if all webhooks in the old object provided recognized versions
* change validateAdmissionReviewVersions to take `requireRecognizedVersion bool` and pass in hadRecognizedVersions here, and pass in `true` from `ValidateMutatingWebhookConfiguration` (you did something similar in validateCustomResourceDefinitionSpec and it worked well)